### PR TITLE
update solc to .5, adding unit tests, removing workbench dependency 

### PR DESCRIPTION
**Pull request by alibhujwala**
-     Add unit tests
-     Remove WorkBench base class since it is WB specific
-     Update some sol styles according to Solidity style guide
-     Add "truffle-assertions" to test reverts and emit for unit test

### CODE REVIEW

File: test/basicprovenance.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const BasicProvenance = artifacts.require("BasicProvenance")
+const truffleAssert = require('truffle-assertions');
+
+contract('BasicProvenance', accounts => {
+    const owner = accounts[0];
+    const observer = accounts[1];
+    const new_owner = accounts[2];
```
Comment (theMobiusStrip) [inline test/basicprovenance.js]: stick to consistent naming style.
i.e. ```newOwner```

File: test/basicprovenance.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        assert.equal(supplyChainOwner, owner, 'owner not correctly set');
+        assert.equal(supplyChainObserver, observer, 'observer not correctly set');
+    });
+
+    it('nonowner can not transfer/complete', async() => {
+        const basicProvenance = await BasicProvenance.deployed();
+
+        truffleAssert.reverts(basicProvenance.TransferResponsibility(owner, { from: new_owner}));
+        truffleAssert.reverts(basicProvenance.Complete({ from: new_owner}));
+
```
Comment (theMobiusStrip) [inline test/basicprovenance.js]: nit: remove blank lines

File: test/basicprovenance.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    it('owner can transfer', async() => {
+        const basicProvenance = await BasicProvenance.deployed();
+
+        await basicProvenance.TransferResponsibility(new_owner);
+        const newCounterParty = await basicProvenance.Counterparty(); 
+
+        assert.equal(newCounterParty, new_owner, 'responsibility not transferred successfully');
```
Comment (theMobiusStrip) [inline test/basicprovenance.js]: you can also consider adding a truffle assert to test emitted events

**Comment by caleteeter** [comment]
Please add the appropriate package.json to the project, its needed for truffle-hdwallet-provider.

**Comment by caleteeter** [comment]
Also, can we add a few more tests.  For instance, we should have a test that the initiating counterparty cannot transfer to itself.  Check for reverts as well.

File: contracts/BasicProvenance.sol [hunk 4]
```diff
@@ -1,77 +1,62 @@
...
-	function BasicProvenance(address supplyChainOwner, address supplyChainObserver) public
-	{
+
+    event LogContractCreated(string applicationName, string workflowName, address originatingAddress);
+    event LogContractUpdated(string applicationName, string workflowName, string action, address originatingAddress);
+
+    string internal ApplicationName;
+    string internal WorkflowName;
+
+    function ContractCreated() private {
```
Comment (caleteeter) [inline contracts/BasicProvenance.sol]: Is there a need for this function when the calling function can emit the event directly?

File: contracts/BasicProvenance.sol [hunk 5]
```diff
@@ -1,77 +1,62 @@
...
+        emit LogContractCreated(ApplicationName, WorkflowName, msg.sender);
+    }
+
+    function ContractUpdated(string memory action) private {
```
Comment (caleteeter) [inline contracts/BasicProvenance.sol]: Is there a need for this function when the calling function can emit the event directly?

**Comment by caleteeter** [comment]
Additionally, can you update the truffle box configuration to run `npm install` as unbox hook.

**Pull request merged**